oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/UserAgentImpl.java

```diff
@@ -116,6 +144,15 @@ AuthorizationResponse encode(final String methodName, final String... parameters
         }
     }
 
+    static void relayProperties(final Properties properties, final Set<String> propertyNames, final List<String> destinationCommand) {
+        for (final String propertyName : propertyNames) {
+            final String propertyValue = properties.getProperty(propertyName);
+            if (propertyValue != null) {
+                destinationCommand.add("-D" + propertyName + "=" + propertyValue);

```

Comment:
I think this looks good for now.  I am still unclear about how to work with a proxy that requires authentication.  Does passing "username:password@host" as the proxyHost work?

From what I read, it involves using `Authenticator`.  We can probably worry about that when there is a request.